client/src/pages/PostSolution.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, ChangeEvent, FormEvent } from "react";
+import { useParams } from "react-router-dom";
+import { Col, Container, Form, Row } from "react-bootstrap";
+import Button from "../components/Button.tsx";
+
+interface FormData {
+  title: string;
+  description: string;
+  diagram: File | null;
+}
+
+// TODO: Add form validation
+// TODO: Include data about the challenge like title
```

Comment:
Challenge title should be added by the user. The should be advised on what include in the solution title (e.g. challenge name + design patterns). The solution is tied to a challenge based on id. 

We can add a line after the solution is posted. For example `solution to XYZ`

Reply:
I want to limit the amount of requests made per page in sequence for keeping load times reasonable. Can we include all the information necessary to display the solution card in the solution table. We can copy over the title from challenge.